zanata-client-commands/src/main/java/org/zanata/client/commands/push/RawPushCommand.java

```diff
@@ -75,10 +94,164 @@ public RawPushCommand(PushOptions opts, RestClientFactory clientFactory) {
+                continue;
+            }
+
+            //throw error if file type is input more than once
+            if (filteredFileTypes.containsKey(inputFileType)) {
+                String msg =
+                    "File type '" + fileType + "' cannot be used more than once in option --file-types.";
+                log.error(msg);
+                throw new RuntimeException(msg);
+            }
+
+            Set<String> filteredExtensions = new HashSet<>(extensions);
+
+            if (!extensions.isEmpty()) {
+                //throw exception if user input extension if not part of DocumentType supported source extension
+                for (String extension: extensions) {
+                    if (!inputFileType.getSourceExtensions()
+                            .contains(extension)) {
+                        String msg =
+                                "Extension '"
+                                        + extension
+                                        + "' for file type '"
+                                        + fileType
+                                        + "' is not supported by the target server and will be ignored.";
+                        consoleInteractor.printfln(DisplayMode.Warning, msg);
+                        filteredExtensions.remove(extension);
+                    }
+                }
+            }
+
+            /**
+             * Use the extensions from typeWithExtension input if exists,
+             * otherwise, use the extensions from server.
+             */
+            filteredExtensions =
+                filteredExtensions.isEmpty() ? inputFileType.getSourceExtensions()
+                            : filteredExtensions;
+
+            //throw error if same file extension found in multiple file type
+            for (DocumentType filteredDocType: filteredFileTypes.keySet()) {
+                for (String filteredExtension: filteredDocType.getSourceExtensions()) {
+                    if (filteredExtensions.contains(filteredExtension)) {
+                        String msg =
+                                "Conflict file type between '" + fileType
+                                        + "' and '" + filteredDocType.name() + "' "
+                                        + "for extension '"
+                                        + filteredExtension
+                                        + "' in option --file-types.";
+                        log.error(msg);
+                        throw new RuntimeException(msg);
+                    }
+                }
+            }
+            filteredFileTypes.put(inputFileType, filteredExtensions);
+        }
+        return filteredFileTypes;
+    }
+
     @Override
     public void run() throws IOException {
         PushCommand.logOptions(log, getOpts());
-        log.warn("Using EXPERIMENTAL project type 'file'.");
+        consoleInteractor.printfln(DisplayMode.Warning,
+            "Using EXPERIMENTAL project type 'file'.");
```

Comment:
Oh.. must have removed it by mistake. 
It should be still experimental.

Regards, 

Alex Eng 
Senior Software Engineer 
DID: +61 3514 8262 
Mobile: +614 2335 3457 
Email: aeng@redhat.com 

Red Hat, Asia-Pacific Pty Ltd 
Level 1, 193 North Quay 
Brisbane 4000 
Office: +61 7 3514 8100 
Fax: +61 7 3514 8199 
Website: www.redhat.com 

----- Original Message -----
From: "David Mason" <notifications@github.com>
To: "zanata/zanata-client" <zanata-client@noreply.github.com>
Cc: "Alex Eng" <aeng@redhat.com>
Sent: Friday, 1 May, 2015 1:41:10 PM
Subject: Re: [zanata-client] Implement doc type support for file projec type (#57)

>      @Override
>      public void run() throws IOException {
>          PushCommand.logOptions(log, getOpts());
> -        log.warn("Using EXPERIMENTAL project type 'file'.");
> +        consoleInteractor.printfln(DisplayMode.Warning,
> +            "Using EXPERIMENTAL project type 'file'.");

- [ ] @aeng it looks like you stopped calling it experimental. Do our QA/UX guys think it is stable and feature-complete enough for that?

---
Reply to this email directly or view it on GitHub:
https://github.com/zanata/zanata-client/pull/57/files#r29488371